src/providers/cambridge/utils.py

```diff
@@ -0,0 +1,31 @@
+import requests
+
+def get_page_content(url):
+    try:
+        response = requests.get(url)
+        response.raise_for_status()  # Raise an HTTPError if the response status code indicates an error
+        return response.text
+    except requests.exceptions.HTTPError as err:
+        print(f"HTTP Error: Failed to fetch the webpage ({err})")
+        return None
```

Comment:
No need to return None. Function without return statement, by default, returns None:
https://realpython.com/python-return-statement/#implicit-return-statements